### PR TITLE
OOPS using Java

### PR DESCRIPTION
**Pull request by Suryac72**
### Covering OOPS using Java

- [x] Added OOPS Concept using Java

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/dev-madhurendra/OOPs/blob/main/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one file. To ease review, please open separate PRs for separate concepts.
- [x] All new folders are placed in an existing directory.
- [x] All new files are placed inside an existing directory.
- [x] All folder names should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`Golang` is allowed but `golang` is not
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`Class.cpp` is allowed but `classe.cpp` is not
- [x] All new concepts have a URL in their comments that points to Wikipedia or another similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`

### CODE REVIEW

**Comment by dev-madhurendra** [comment]
Hi @Suryac72 , Please read the [contribution guidelines](https://github.com/dev-madhurendra/OOPs/blob/main/CONTRIBUTING.md)
- One concept per PR
- Naming Convention for branch name,folder name,file name etc.

**Comment by Suryac72** [comment]
PR updated (One concept one PR)

File: Java/Classes.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+
+/**
+ * OOPs (Object-Oriented Programming System)
+ * 
+ * Object means a real-world entity such as a pen, chair, table, computer,
```
Comment (dev-madhurendra) [inline Java/Classes.java:5]: - Class definition is missing .
- No need of object definition here 
- Particular file should contains only particular definition.

File: Java/Classes.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * watch, etc.
+ * Object-Oriented Programming is a methodology or paradigm to design a program
+ * using classes and objects.
+ * It simplifies software development and maintenance by providing some
+ * concepts:
+ * 
+ * Object
+ * Class
+ * Inheritance
+ * Polymorphism
```
Comment (dev-madhurendra) [inline Java/Classes.java:15]: - Create a new .txt file and add common things there only.

File: Java/Classes.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+class Vehicle {
+    /**
+     * OBJECTS
+     * -----------------------------------------------------
+     * Any entity that has state and behavior is known as an object.
+     * For example, a chair, pen, table, keyboard, bike, etc.
+     * It can be physical or logical.
+     * 
+     * An Object can be defined as an instance of a class.
+     * An object contains an address and takes up some space in memory.
```
Comment (dev-madhurendra) [inline Java/Classes.java:38]: - No need object definition here again

File: Java/Classes.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    //data members, private in nature
+    private String vehicleNumber;
+    private float price;
+    private String model;
+    private String color;
+    private String variant;
+    private String vehicleName;
+
+    public String getVehicleNumber() {
```
Comment (dev-madhurendra) [inline Java/Classes.java:55]: - Use these access modifier in the access modifier at the time of that concepts only.
- for now you can use public

File: Java/Classes.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    public String getVehicleName() {
+        return vehicleName;
+    }
+
+    public void setVehicleName(String vehicleName) {
+        this.vehicleName = vehicleName;
+    }
+
```
Comment (dev-madhurendra) [inline Java/Classes.java:102]: - Don't add too many getter and setter method here.

File: Java/Classes.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    public static void main(String args[]) {
+        Vehicle car = new Vehicle(); //creating car object from its blueprint class
+        car.setVehicleName("TATA PUNCH");
+        car.setColor("Yellow");
+        car.setModel("MX11");
+        car.setPrice(1000000);
+        car.setVariant("LX-11");
+        car.setVehicleNumber("1S23DDE");
+
+        System.out.println(car.toString());
```
Comment (dev-madhurendra) [inline Java/Classes.java:121]: - What will be the expected output here add those also

**Comment by dev-madhurendra** [comment]
@Suryac72  There should be dedicated folder for every concepts like e.g. `for class add folder like Java/Classes and then place file name here and make sure file should not duplicate.`

**Review by dev-madhurendra — CHANGES_REQUESTED** [review]

File: Java/Classes.java (hunk 6 above)
Comment (Suryac72) [reply Java/Classes.java]: hi @dev-madhurendra, PR updated as per the mentioned comments. kindly review it

File: Java/Classes/Classes.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
+package Classes;
+/**
+ * Class : Class is a blueprint which defines how our real world entity will be represented in terms of code.
+ *
```
Comment (dev-madhurendra) [inline Java/Classes/Classes.java:4]: Add more authentic definition.

Comment (dev-madhurendra) [reply Java/Classes/Classes.java:4]: Otherwise everything looks good to me.

**Review by dev-madhurendra — CHANGES_REQUESTED** [review]

File: Java/Classes/Classes.java (hunk 7 above)
Comment (Suryac72) [reply Java/Classes/Classes.java:4]: @dev-madhurendra  PR updated

**Review by dev-madhurendra — APPROVED** [review]

**Comment by dev-madhurendra** [comment]
Hi @Suryac72 , I am approving this and merging this but please explain every topic in a proper manner so that anyone who is reading can easily understand concepts.

**Comment by Suryac72** [comment]
Not able to merge, needs 2 approvals

**Pull request merged**